client/network/src/protocol/block_requests.rs

```diff
@@ -192,7 +192,7 @@ where
 
 		let mut blocks = Vec::new();
 		let mut block_id = from_block_id;
-		while let Some(header) = self.chain.header(&block_id).unwrap_or(None) {
+		while let Some(header) = self.chain.header(block_id).unwrap_or(None) {
```

Comment:
Why does `fn header` take ownership over `BlockId`, but methods in `BlockProvider` don't? Can't we unify this and use references everywhere?

Reply:
yeah so this method comes from `HeaderBackend`, This would require refactoring the trait itself and it's usage. Happy to do this in a separate PR.

Reply:
Fine with me!